src/ChainSafe.GamingSdk.Gelato/Gelato.cs

```diff
@@ -0,0 +1,262 @@
+                };
+
+                var optional = await CallWithErc2771RequestOptionalParameters.PopulateOptionalUserParameters(callRequest, Erc2771Type.CallWithSyncFee, config, chainConfig, contractBuilder);
+                var newStruct = callRequest.MapRequestToStruct<CallWithSyncFeeErc2771Struct>(optional, Erc2771Type.CallWithSyncFee);
+
+                callRequest.UserDeadline ??= optional.UserDeadline;
+                callRequest.UserNonce ??= optional.UserNonce;
+
+                var types = new Dictionary<string, MemberDescription[]>
+                {
+                    ["CallWithSyncFeeERC2771"] = new[]
+                    {
+                        new MemberDescription { Name = "chainId", Type = "uint256" },
+                        new MemberDescription { Name = "target", Type = "address" },
+                        new MemberDescription { Name = "data", Type = "bytes" },
+                        new MemberDescription { Name = "user", Type = "address" },
+                        new MemberDescription { Name = "userNonce", Type = "uint256" },
+                        new MemberDescription { Name = "userDeadline", Type = "uint256" },
+                    },
+                };
+
+                callRequest.Signature = await signer.SignTypedData(GetEip712Domain(Erc2771Type.CallWithSyncFee), types, newStruct);
+
+                return await gelatoClient.Post<CallWithErc2771Request, RelayResponse>(RelayCall.CallWithSyncFeeErc2771, callRequest);
+            }
+            catch (Exception e)
+            {
+                throw new Web3Exception($"GelatoRelaySDK/relayWithSyncFee: Failed with error: {e.Message}");
+            }
+        }
+
+        public async Task<RelayResponse> SponsoredCall(SponsoredCallRequest request)
+        {
+            if (config.SponsorApiKey == null)
+            {
+                throw new Web3Exception("GelatoRelaySDK/sponsoredCall: Sponsor api key not provided");
+            }
+
+            try
+            {
+                request.SponsorApiKey = config.SponsorApiKey;
+                request.ChainId = int.Parse(chainConfig.ChainId);
+                return await gelatoClient.Post<SponsoredCallRequest, RelayResponse>(RelayCall.SponsoredCall, request);
+            }
+            catch (Exception e)
+            {
+                throw new Web3Exception($"GelatoRelaySDK/sponsoredCall: Failed with error: {e.Message}");
+            }
+        }
+
+        public async Task<RelayResponse> SponsoredCallErc2771(SponsoredCallErc2771Request request)
+        {
+            return await SponsoredCallErc2771(request, null);
+        }
+
+        // TODO: consume options
+        public async Task<RelayResponse> SponsoredCallErc2771(SponsoredCallErc2771Request request, RelayRequestOptions options)
+        {
+            if (signer == null)
+            {
+                throw new Web3Exception("GelatoRelaySDK/SponsoredCallErc2771: No signer present for request");
```

Comment:
Same here

Reply:
@oleksandrchainsafe Same here what?

Reply:
"We could point out, that the signer should be registered during the build time, to make it more obvious."